generator/js_generator.cc

```diff
@@ -980,7 +985,7 @@ bool DeclaredReturnTypeIsNullable(const GeneratorOptions& options,
     return false;
   }
 
-  if (field->file()->syntax() == FileDescriptor::SYNTAX_PROTO3 &&
+  if (FileDescriptorLegacy(field->file()).syntax() == FileDescriptorLegacy::Syntax::SYNTAX_PROTO3 &&
```

Comment:
What are the legacy descriptors needed for? I haven't seen them used before and the inclusion the descriptor_legacy.h seems to be causing a build issue.

Reply:
I think that `field->file()->syntax()` is deprecated and no longer available in the FileDescriptor class. Using FileDescriptorLegacy seemed like the easiest and lowest-risk fix since I'm not entirely sure what this is actually trying to check. Similar concerns showed up a couple other places.